tools/topology/sof-tgl-sdw-max98373-rt5682.m4

```diff
@@ -53,9 +53,9 @@ define(`SMART_BE_ID', 2)
 # Playback related
 define(`SMART_PB_PPL_ID', 3)
 define(`SMART_PB_CH_NUM', 2)
-define(`SMART_TX_CHANNELS', 4)
-define(`SMART_RX_CHANNELS', 8)
-define(`SMART_FB_CHANNELS', 8)
+define(`SMART_TX_CHANNELS', 2)
+define(`SMART_RX_CHANNELS', 4)
+define(`SMART_FB_CHANNELS', 4)
```

Comment:
that's still too much, we can only support 4 separate transfers PCM on a single link.

Reply:
@plbossart  I tested it on the TGL production.  And 4 separate PCM transfers : two channels playback & two channels capture ?  As you know,  the channel number for capture is double size of playback for IV feedback.

Reply:
@plbossart  yes, it is impossible for 32bit * 6.   32bit PCM stream is used in DSM feature, but DAI config is set to 24bit (another story), so my PR works.

Reply:
This has nothing to do with bandwidth, word length or channel count. At the ALH level, we can only do 4 DMA transfers to/from PDI_BD PCM targets.

```c
#define DAI_NUM_ALH_BI_DIR_LINKS_GROUP	6

static const uint8_t alh_handshake_map[64] = {
	-1,	/* 0  - INVALID */
	-1,	/* 1  - INVALID */
	-1,	/* 2  - INVALID */
	-1,	/* 3  - INVALID */
	-1,	/* 4  - INVALID */
	-1,	/* 5  - INVALID */
	-1,	/* 6  - INVALID */
	22,	/* 7  - BIDIRECTIONAL */
	23,	/* 8  - BIDIRECTIONAL */
	24,	/* 9  - BIDIRECTIONAL */
	25,	/* 10 - BIDIRECTIONAL */
	26,	/* 11 - BIDIRECTIONAL */
	27,	/* 12 - BIDIRECTIONAL */
````

We can only use e.g. 9..12 for PCM transfers. I am not sure how you managed to transfer 6 independent streams since it's not aligned with the documentation. Can you describe which ALH stream numbers you used?

Reply:
@plbossart  Iet's make it clear.  There are only two PCMs in DSM feature:  one is for playback (2ch) and another one is for capture (4ch).

Reply:
I was talking about ALH/DMA transfers of PCM data, your reply does not answer to my question.

Reply:
@plbossart  I don't need to transfer 6 independent streams, right ? There are only two streams, so 22,	/* 7  - BIDIRECTIONAL */ is used for playback and 23,	/* 8  - BIDIRECTIONAL */ is used for capture.

Reply:
I dumped the stream id in driver,   32  /* 23 - BIDIRECTIONAL / is used for playback and 33 /* 24 - BIDIRECTIONAL / is used for capture ( link1).

Reply:
Got it now from the comments, best to say "FEEDBACK" instead of "FB" in the naming.